cc1120_arduino/cc1120_spi.c

```diff
@@ -50,40 +53,50 @@ bool cc1120_read_spi(uint8_t addr, uint8_t data[], uint8_t len) {
  * @param addr - The address of the first register to read.
  * @param data - The array to store the read data, or a pointer to a single uint8_t if len=1.
  * @param len - The number of registers to read.
- * @return true - If the read was successful.
- * @return false - If the register is not valid, or the status byte is invalid.
+ * @return CC1120_ERROR_CODE_SUCCESS - If the read was successful.
+ * @return An error code - If the register is not valid, or the status byte is invalid.
  */
-bool cc1120_read_ext_addr_spi(uint8_t addr, uint8_t data[], uint8_t len) {
-    bool status = true;
+cc1120_error_code cc1120_read_ext_addr_spi(uint8_t addr, uint8_t data[], uint8_t len) {
+    cc1120_error_code status = CC1120_ERROR_CODE_SUCCESS;
 
-    if (addr > CC1120_REGS_EXT_PA_CFG3 && addr < CC1120_REGS_EXT_WOR_TIME1 ||
-        addr > CC1120_REGS_EXT_XOSC_TEST0 && addr < CC1120_REGS_EXT_RXFIRST ||
-        addr > CC1120_REGS_EXT_FIFO_NUM_RXBYTES) {
+    if ((addr > CC1120_REGS_EXT_PA_CFG3 && addr < CC1120_REGS_EXT_WOR_TIME1) ||
+        (addr > CC1120_REGS_EXT_XOSC_TEST0 && addr < CC1120_REGS_EXT_RXFIRST) ||
+        (addr > CC1120_REGS_EXT_FIFO_NUM_RXBYTES)) {
         mcu_log(CC1120_LOG_LEVEL_ERROR, "cc1120_read_ext_addr_spi: Not a valid register!\n");
-        status = false;
+        status = CC1120_ERROR_CODE_READ_EXT_ADDR_SPI_INVALID_REGISTER;
+        return status;
     }
 
     if (len < 1) {
         mcu_log(CC1120_LOG_LEVEL_ERROR, "cc1120_read_ext_addr_spi: Not a valid length!\n");
-        status = false;
+        status = CC1120_ERROR_CODE_READ_EXT_ADDR_SPI_INVALID_LENGTH;
+        return status;
     }
     
 
-    if (status) {
-        uint8_t header = (len > 1) ? R_BIT | BURST_BIT | CC1120_REGS_EXT_ADDR :
-                                     R_BIT | CC1120_REGS_EXT_ADDR;
+    if (status == CC1120_ERROR_CODE_SUCCESS) {
+        uint8_t header = (len > 1) ? (R_BIT | BURST_BIT | CC1120_REGS_EXT_ADDR) :
+                                     (R_BIT | CC1120_REGS_EXT_ADDR);
 
         mcu_cc1120_cs_assert();
-        if (!cc1120_send_byte_receive_status(header)) {
-            status = false;
+        if (cc1120_send_byte_receive_status(header) != CC1120_ERROR_CODE_SUCCESS) {
+            status = CC1120_ERROR_CODE_READ_EXT_ADDR_SPI_INVALID_STATUS_BYTE;
```

Comment:
Can merge the different "INVALID_STATUS_BYTE" errors into one, I don't think we need one from every SPI function
If we wanted to know which function returned this, it would probably be better and easier to implement some method to read the function call stack (like a bunch of logging calls)